sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/ThetaSketchUtilsSuite.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+import org.apache.spark.{SparkFunSuite, SparkRuntimeException}
+
+class ThetaSketchUtilsSuite extends SparkFunSuite {
+
+  test("checkLgNomLongs: accepts values within valid range") {
+    val validValues =
+      Seq(ThetaSketchUtils.MIN_LG_NOM_LONGS, 10, 20, ThetaSketchUtils.MAX_LG_NOM_LONGS)
+    validValues.foreach { value =>
+      // Should not throw
+      ThetaSketchUtils.checkLgNomLongs(value)
+    }
+  }
+
+  test("checkLgNomLongs: throws exception for values below minimum") {
+    val invalidValues = Seq(ThetaSketchUtils.MIN_LG_NOM_LONGS - 1, 0, -5)
+    invalidValues.foreach { value =>
+      val e = intercept[SparkRuntimeException] {
+        ThetaSketchUtils.checkLgNomLongs(value)
+      }
+      assert(
+        e.getMessage.contains(
```

Comment:
Can you please switch these unit tests to use `checkError` instead of asserting on substrings of the error message? Example: [1].

[1] https://github.com/apache/spark/blob/596d03fc541cd9433fdfd511e131aa1cbc603a85/sql/core/src/test/scala/org/apache/spark/sql/connector/FileDataSourceV2FallBackSuite.scala#L101-L107

Reply:
yes, thanks for pointing this out